game/core/game_state.py

```diff
@@ -76,24 +76,54 @@ def save_player_data(self, player) -> None:  # type: ignore[no-untyped-def]
             pickle.dump(data, _f)
 
     @beartype
-    def remove_sprite_from_map(self, sprite, searchable: bool = False) -> None:  # type: ignore[no-untyped-def]
-        sprite_id = sprite.properties["id"]
+    def remove_sprite_from_map(
+        self, removed_sprite: Sprite, searchable: bool = False
+    ) -> None:
+        sprite_id = removed_sprite.properties["id"]
         index = self.searchable_index if searchable else self.tree_index
         layer_copy = self.tile_map["layers"][index]
 
-        obj_to_remove = None
-        for obj in layer_copy["objects"]:
-            for prop in obj["properties"]:
-                if prop["name"] == "id" and prop["value"] == sprite_id:
-                    obj_to_remove = obj
-                    break
-        if obj_to_remove:
+        if obj_to_remove := next(
+            (
+                obj
+                for obj in layer_copy["objects"]
+                for prop in obj["properties"]
+                if prop["name"] == "id" and prop["value"] == sprite_id
+            ),
+            None,
+        ):
             layer_copy["objects"].remove(obj_to_remove)
             self.tile_map["layers"][index] = layer_copy
 
-        sprite.properties["removed"] = True
-        sprite.remove_from_sprite_lists()
-        self.save_map_data()
+            removed_sprite.properties["removed"] = True
+            removed_sprite.remove_from_sprite_lists()
+
+            if item_drop := removed_sprite.properties.get("drop"):
+                new_sprite = arcade.Sprite(f":assets:{item_drop}.png")
+                new_sprite.properties = {"name": item_drop, "id": str(uuid4())}
+                new_gid = obj_to_remove["gid"] + 10_000
+                new_obj = {
+                    "id": obj_to_remove["id"] + 1_000,
+                    "gid": new_gid,
+                    "height": new_sprite.height,
+                    "name": "",
+                    "properties": [
+                        {"name": key, "type": "string", "value": value}
+                        for key, value in new_sprite.properties.items()
+                    ],
+                    "rotation": 0,
+                    "visible": True,
+                    "width": new_sprite.width,
+                    "x": obj_to_remove["x"],
+                    "y": obj_to_remove["y"],
+                }
+                index = self.searchable_index
+                self.tile_map["layers"][index]["objects"].append(new_obj)
+                self.tile_map["tilesets"].append(
+                    {"firstgid": new_gid, "source": f"{item_drop}.json"}
```

Comment:
Note: I'm reworking some changes locally to get the Wood sprite on the Map without a reload (Still WIP)